src/ArangoDBSnapshotStore.php

```diff
@@ -0,0 +1,135 @@
+ * (c) 2016-2016 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2016 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\ArangoDB\SnapshotStore;
+
+use ArangoDBClient\Batch;
+use ArangoDBClient\Connection;
+use ArangoDBClient\Statement;
+use Prooph\EventSourcing\Aggregate\AggregateType;
+use Prooph\EventSourcing\Snapshot\Snapshot;
+use Prooph\EventSourcing\Snapshot\SnapshotStore;
+
+final class ArangoDBSnapshotStore implements SnapshotStore
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * Custom sourceType to snapshot mapping
+     *
+     * @var array
+     */
+    private $snapshotCollectionMap;
+
+    /**
+     * @var string
+     */
+    private $defaultSnapshotCollectionName;
+
+    public function __construct(
+        Connection $connection,
+        array $snapshotCollectionMap = [],
+        string $defaultSnapshotCollectionName = 'snapshots'
+    ) {
+        $this->connection = $connection;
+        $this->snapshotCollectionMap = $snapshotCollectionMap;
+        $this->defaultSnapshotCollectionName = $defaultSnapshotCollectionName;
+    }
+
+    public function get(AggregateType $aggregateType, string $aggregateId): ?Snapshot
+    {
+        $collectionName = $this->getCollectionName($aggregateType);
+        $query = <<<EOF
+FOR s IN @@collection 
+    FILTER s._key == @aggregate_id && s.aggregate_type == @aggregate_type
+    SORT s.last_version DESC 
+    RETURN s
+EOF;
+
+        $statement = new Statement(
+            $this->connection, [
+                'query' => $query,
+                'bindVars' => [
+                    '@collection' => $collectionName,
+                    'aggregate_type' => $aggregateType->toString(),
+                    'aggregate_id' => $aggregateId,
+                ],
+            ]
+        );
+
+        $cursor = $statement->execute();
+
+        if (! $cursor->getCount()) {
+            return null;
+        }
+
+        $result = $cursor->current();
+
+        return new Snapshot(
+            $aggregateType,
+            $aggregateId,
+            unserialize($result->aggregate_root),
+            (int) $result->last_version,
+            \DateTimeImmutable::createFromFormat('Y-m-d\TH:i:s.u', $result->created_at, new \DateTimeZone('UTC'))
+        );
+    }
+
+    public function save(Snapshot $snapshot): void
+    {
+        $collectionName = $this->getCollectionName($snapshot->aggregateType());
+
+        $batch = new Batch($this->connection);
+
+        $batch->append(
+            'DELETE',
+            'DELETE /_api/document/' . $collectionName . '/' . $snapshot->aggregateId() . ' HTTP/1.1'
+        );
+
+        $aggregateRoot = $snapshot->aggregateRoot();
+
+        if ($aggregateRoot instanceof \JsonSerializable) {
```

Comment:
oh that's nice, can you create PR and add that for mongodb- & pdo-snapshot-store, too?